crates/sui-framework/packages/sui-framework/sources/kiosk/kiosk.move

```diff
@@ -412,6 +415,181 @@ module sui::kiosk {
+        /// into a single dynamic field, we reduce the amount of fields on the
+        /// top level (eg items / listings) while giving extension developers
+        /// the ability to store any data they want.
+        storage: Bag,
+        /// Bitmask of permissions that the extension has (can be revoked any
+        /// moment). It's all or nothing policy - either the extension has the
+        /// required permissions or no permissions at all.
+        permissions: u32,
+        /// List of types that the extension is allowed to use. It only affects
+        /// permissioned actions (eg `place`, `lock`). The Kiosk owner can limit
+        /// an extensions functionality to a set of specified types.
+        types: vector<TypeName>,
+        /// Whether the requested permissions are still available. While the
+        /// owner can't delete the `Bag` storage unless the extension provides
+        /// the functionality, they can revoke the permissions at any time,
+        /// making the extension a harmless storage - similar to Kiosk having an
+        /// exposed UID.
+        is_enabled: bool,
+    }
+
+    /// The `ExtensionKey` is a typed dynamic field key used to store the
+    /// extension configuration and data. `Ext` is a phantom type that is used
+    /// to identify the extension witness.
+    struct ExtensionKey<phantom Ext> has store, copy, drop {}
+
+    /// Add an extension to the Kiosk. Can only be performed by the owner.
+    ///
+    /// Unlike the original implementation, this one uses the `public`
+    /// visibility modifier to allow "owned kiosk" extension. Ideally we don't
+    /// want this function to be called arbitrarily, and to prevent some
+    /// malicious scenarios we now require the extension witness on install.
+    public fun add_extension<Ext: drop>(
+        _ext: Ext,
+        self: &mut Kiosk,
+        cap: &KioskOwnerCap,
+        permissions: u32,
+        types: vector<TypeName>,
+        ctx: &mut TxContext
+    ) {
+        assert!(object::id(self) == cap.for, ENotOwner);
+
+        df::add(&mut self.id, ExtensionKey<Ext> {}, Extension {
+            storage: bag::new(ctx),
+            permissions,
+            types,
+            is_enabled: true,
+        })
+    }
+
+    /// Revoke permissions from the extension. While it does not remove the
+    /// extension completely, it keeps it from performing any protected actions.
+    /// The storage is still available to the extension (until it's removed).
+    public fun disable_extension<Ext: drop>(
+        self: &mut Kiosk,
+        cap: &KioskOwnerCap,
+    ) {
+        assert!(object::id(self) == cap.for, ENotOwner);
+        extension_mut<Ext>(self).is_enabled = false;
+    }
+
+    /// Re-enable permissions for the extension. Can only be performed by the
+    /// owner. The extension can start performing protected actions again.
+    public fun re_enable_extension<Ext: drop>(
```

Comment:
```suggestion
    public fun enable_extension<Ext: drop>(
```

+ noting that extensions are automatically enabled after `add_extension` will be clearer, I think.

Reply:
I didn't quite get that? If the signature is "enable" then it causes confusion around default "enabled" scenario.